tests/e2e/tests/test_kiali_service_endpoint.py

```diff
@@ -1,46 +1,138 @@
-    number_services = len (kiali_client.service_list(namespace=bookinfo_namespace).get('services'))
-    assert (number_services == BOOKINFO_EXPECTED_SERVICES) or (number_services == BOOKINFO_EXPECTED_SERVICES_MONGODB)
+    service_list_json = kiali_client.service_list(namespace=bookinfo_namespace)
+    assert service_list_json.get('namespace').get('name') == bookinfo_namespace
+
+    services = service_list_json.get('services')
+    assert (len (services) == BOOKINFO_EXPECTED_SERVICES) or (len (services) == BOOKINFO_EXPECTED_SERVICES_MONGODB)
+
+    for service in services:
+      assert service != None
+      assert service.get('name') != None and service.get('name') != ''
+      assert service.get('istioSidecar') == True
+      assert service.get('appLabel') == True
 
 def test_service_detail_endpoint(kiali_client):
     bookinfo_namespace = conftest.get_bookinfo_endpoint()
 
     service_details = kiali_client.service_details(namespace=bookinfo_namespace, service=SERVICE_TO_VALIDATE)
     assert service_details != None
     assert service_details.get('istioSidecar') == True
+    assert 'ports' in service_details.get('service')
+    assert 'labels' in service_details.get('service')
     assert 'workloads' in service_details
     assert 'service' in service_details
+    assert 'endpoints' in service_details
+    assert 'virtualServices' in service_details
+    assert 'destinationRules' in service_details
+    assert 'health' in service_details
+
+def test_service_detail_with_virtual_service(kiali_client):
+    bookinfo_namespace = conftest.get_bookinfo_endpoint()
+
+    # Add a virtual service that will be tested
+    assert command_exec.oc_apply(VIRTUAL_SERVICE_FILE, bookinfo_namespace) == True
+
+    service_details = kiali_client.service_details(namespace=bookinfo_namespace, service=SERVICE_TO_VALIDATE)
+    assert service_details != None
+
+    virtual_service = service_details.get('virtualServices')[0]
+    assert virtual_service != None
+    assert virtual_service.get('name') == 'reviews'
+
+    https = virtual_service.get('http')
+    assert https != None
+    assert len (https) == 1
+
+    routes = https[0].get('route')
+    assert len (routes) == 2
+
+    assert routes[0].get('weight') == 80
+    destination = routes[0].get('destination')
+    assert destination != None
+    assert destination.get('host') == 'reviews'
+    assert destination.get('subset') == 'v1'
+
+    assert routes[1].get('weight') == 20
+    destination = routes[1].get('destination')
+    assert destination != None
+    assert destination.get('host') == 'reviews'
+    assert destination.get('subset') == 'v2'
+
+    assert command_exec.oc_delete(VIRTUAL_SERVICE_FILE, bookinfo_namespace) == True
+
+def test_service_detail_with_destination_rule(kiali_client):
+    bookinfo_namespace = conftest.get_bookinfo_endpoint()
+
+    # Add a virtual service that will be tested
+    assert command_exec.oc_apply(DESTINATION_RULE_FILE, bookinfo_namespace) == True
+
```

Comment:
Similar question as to the above VS timing possibility - could there be false failures below due to the rule not yet being propagated through Istio?